src/core/lib.lua

```diff
@@ -373,7 +373,8 @@ if ffi.abi("be") then
    function htonl(b) return b end
    function htons(b) return b end
 else
-   function htonl(b) return bswap(b) end
+  -- htonl is unsigned, like the C version, to avoid traps...
+   function htonl(b) return tonumber(ffi.cast(uint32_t, bswap(b))) end
```

Comment:
Excellent catch. I'm so used to dealing with the results of whichevertype_t = ffi.typeof(....) that I entirely overlooked that, but it does indeed make the tests pass locally. Uses of it will still need to be manually checked, though, since this is a breaking semantic change.